lib/util/stream_utils.js

```diff
@@ -38,103 +39,148 @@ shaka.util.StreamUtils = class {
       preferredAudioCodecs, preferredAudioChannelCount,
       preferredDecodingAttributes) {
     const StreamUtils = shaka.util.StreamUtils;
+    const MimeUtils = shaka.util.MimeUtils;
 
     let variants = manifest.variants;
-
     // To start, choose the codecs based on configured preferences if available.
     if (preferredVideoCodecs.length || preferredAudioCodecs.length) {
       variants = StreamUtils.choosePreferredCodecs(variants,
           preferredVideoCodecs, preferredAudioCodecs);
     }
 
-    // Consider a subset of variants based on audio channel
-    // preferences.
-    // For some content (#1013), surround-sound variants will use a different
-    // codec than stereo variants, so it is important to choose codecs **after**
-    // considering the audio channel config.
-    variants = StreamUtils.filterVariantsByAudioChannelCount(
-        variants, preferredAudioChannelCount);
-
-    // Now organize variants into buckets by codecs.
-    /** @type {!shaka.util.MultiMap.<shaka.extern.Variant>} */
-    let variantsByCodecs = StreamUtils.getVariantsByCodecs_(variants);
-    variantsByCodecs = StreamUtils.filterVariantsByDensity_(variantsByCodecs);
+    if (preferredDecodingAttributes.length) {
+      // group variants by resolution and choose preferred variants only
+      /** @type {!shaka.util.MultiMap.<shaka.extern.Variant>} */
+      const variantsByResolutionMap = new shaka.util.MultiMap();
+      for (const variant of variants) {
+        variantsByResolutionMap
+            .push(String(variant.video.width || 0), variant);
+      }
+      const bestVariants = [];
+      variantsByResolutionMap.forEach((width, variantsByResolution) => {
+        let highestMatch = 0;
+        let matchingVariants = [];
+        for (const variant of variantsByResolution) {
+          const matchCount = preferredDecodingAttributes.filter(
+              (attribute) => variant.decodingInfos[0][attribute],
+          ).length;
+          if (matchCount>highestMatch) {
```

Comment:
Nit: This if (and the next) should have whitespace around the operators. E.g. `if (matchCount > highestMatch) {` and `} else if (matchCount == highestMatch) {`

Reply:
Done!